compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -872,7 +872,8 @@ impl<'tcx> Printer<'tcx> for SymbolMangler<'tcx> {
             DefPathData::ValueNs(_) => 'v',
             DefPathData::Closure => 'C',
             DefPathData::Ctor => 'c',
-            DefPathData::AnonConst => 'k',
+            DefPathData::AnonConst => 'K',
+            DefPathData::LateAnonConst => 'k',
```

Comment:
I'm not certain changing mangling is the best idea. Is there some update to a demangler to be made before this PR is accepted?

Reply:
I couldn't find where in the demangler code this is used. IIUC, there are only two categories that are checked, 'C' and 'S', the latter not being produced any more. Should I look somewhere else?

https://github.com/rust-lang/rustc-demangle/blob/c5688cfec32d2bd00701836f12beb3560ee015b8/src/v0.rs#L834-L838

Reply:
If it is unused, the demangler will just fail to demangle. Which is fine, you'd just get a weird symbol. We do have a roundtrip assertion checking that all symbols emitted in tests can be demangled back to the original symbol. I'm not quite certain why this is not causing issues, but maybe paths are just handled well enough together that it all works out.